### PR TITLE
sg cloud eph: simplify status

### PR DESCRIPTION
**Pull request by filiphaftek**
1. Make `sg cloud eph` instance status Reason simple string, as CloudAPI will take over and return Reason with job URL - no need to parse Reason anymore.
```sh
go run ./dev/sg cloud eph status --name ff-eph56
⚠️ Running sg with a dev build, following flags have different default value unless explictly set: skip-auto-update, disable-analytics
✅ Ephemeral instance "ff-eph56" status retrieved
Ephemeral instance details:
Name                                     Expires In           Instance status                          Details
ff-eph56                                 4h37m4.79031s        unspecified                              creation task is already running: https://console.cloud.google.com/workflows/workflow/us-central1/create-instance-c31bd1a4ea84/executions?project=michael-test-03
```
2. Allow re-run create if CloudAPI returns status with Reason - it means instance is not fully created yet, so user might re-try create - CloudAPI will ensure more than one create is not running at the same time.

3. Updated printers with GOTO action for each instance details:
```sh
go run ./dev/sg cloud eph list --all
⚠️ Running sg with a dev build, following flags have different default value unless explictly set: skip-auto-update, disable-analytics
☁️ Fetched 10 instances
Name                                     Expires In           Instance status                          Details
andre-eph-1                              30h10m42.989163s     unspecified                              invoke: sg cloud eph status --name andre-eph-1
ff-eph56                                 4h34m43.989154s      unspecified                              invoke: sg cloud eph status --name ff-eph56
```

## Test plan

Unit tests simplified.
E2e against old and new CloudAPI.

### CODE REVIEW

File: dev/sg/internal/cloud/instance.go [hunk 1]
```diff
@@ -181,7 +114,7 @@ type InstanceFeatures struct {
 func newInstanceStatus(src *cloudapiv1.InstanceState) *InstanceStatus {
 	status := InstanceStatus{}
 	var err error
-	status.Reason, err = newStatusReason(src.GetReason())
+	status.Reason = src.GetReason()
 	if err != nil {
```
Comment (burmudar) [inline dev/sg/internal/cloud/instance.go:118]: Can remove this too

**Comment by burmudar** [comment]
@filiphaftek can you also run it with `--raw`?

**Comment by filiphaftek** [comment]
> @filiphaftek can you also run it with `--raw`?

Yes, tested already:
```sh
go run ./dev/sg cloud eph status --name ff-eph56 --raw
⚠️ Running sg with a dev build, following flags have different default value unless explictly set: skip-auto-update, disable-analytics
✅ Ephemeral instance "ff-eph56" status retrieved
Ephemeral instance details:
ID           : src-2edcd1adad848160cacb
Name         : ff-eph56
InstanceType : ephemeral
Environment  : dev
Version      : 5.5.1337
URL          : https://ff-eph56.sgdev.dev
AdminEmail   : filip.haftek@sourcegraph.com
CreatedAt    : n/a
DeletetAt    : n/a
ExpiresAt    : 2024-08-01T16:00:21Z
Project      : src-7b4a1801a8209eeca568
Region       : us-central1
Status       : unspecified
Details      : creation task is already running: https://console.cloud.google.com/workflows/workflow/us-central1/create-instance-87342535848a/executions?project=michael-test-03
```

**Comment by burmudar** [comment]
> > @filiphaftek can you also run it with `--raw`?
> 
> Yes, tested already:
> 
> ```shell
> go run ./dev/sg cloud eph status --name ff-eph56 --raw
> ⚠️ Running sg with a dev build, following flags have different default value unless explictly set: skip-auto-update, disable-analytics
> ✅ Ephemeral instance "ff-eph56" status retrieved
> Ephemeral instance details:
> ID           : src-2edcd1adad848160cacb
> Name         : ff-eph56
> InstanceType : ephemeral
> Environment  : dev
> Version      : 5.5.1337
> URL          : https://ff-eph56.sgdev.dev
> AdminEmail   : filip.haftek@sourcegraph.com
> CreatedAt    : n/a
> DeletetAt    : n/a
> ExpiresAt    : 2024-08-01T16:00:21Z
> Project      : src-7b4a1801a8209eeca568
> Region       : us-central1
> Status       : unspecified
> Details      : creation task is already running: https://console.cloud.google.com/workflows/workflow/us-central1/create-instance-87342535848a/executions?project=michael-test-03
> ```

Should cloud api status not be `progressing` or `creating` instead of unspecified?

**Comment by filiphaftek** [comment]
> > > @filiphaftek can you also run it with `--raw`?
> > 
> > 
> > Yes, tested already:
> > ```shell
> > go run ./dev/sg cloud eph status --name ff-eph56 --raw
> > ⚠️ Running sg with a dev build, following flags have different default value unless explictly set: skip-auto-update, disable-analytics
> > ✅ Ephemeral instance "ff-eph56" status retrieved
> > Ephemeral instance details:
> > ID           : src-2edcd1adad848160cacb
> > Name         : ff-eph56
> > InstanceType : ephemeral
> > Environment  : dev
> > Version      : 5.5.1337
> > URL          : https://ff-eph56.sgdev.dev
> > AdminEmail   : filip.haftek@sourcegraph.com
> > CreatedAt    : n/a
> > DeletetAt    : n/a
> > ExpiresAt    : 2024-08-01T16:00:21Z
> > Project      : src-7b4a1801a8209eeca568
> > Region       : us-central1
> > Status       : unspecified
> > Details      : creation task is already running: https://console.cloud.google.com/workflows/workflow/us-central1/create-instance-87342535848a/executions?project=michael-test-03
> > ```
> 
> Should cloud api status not be `progressing` or `creating` instead of unspecified?

I though unspecified can be easier for us to differentiate vs instance already created, but progressing upgrade etc [here](https://github.com/sourcegraph/controller/pull/1619/files#diff-4fc2520380cba91dfc6b6843bab547f458a77e42c734e8aa869caef8f472610bR145) - but as we do not use unspecified anyway, I will switch back to progressing 👍

**Review by burmudar — APPROVED** [review]
Small comments! thanks for the fix + improvements!

**Pull request merged**